### ISSUE TITLE
ArrayOutOfBoundException, parseFrom(s: Array[Byte]) doesnt materialize object 

### ISSUE THREAD

**Issue by BBartosz** [scalapb/ScalaPB#230]
Hello I've encounter similar issue:

I've got my protobuf generated class: 

```
MyProtoClass.scala
```

Let's say that I want to parse this value:
```scala
arrayOfBytes: Array[Byte]
```
 I expect this parsing to fail, so I write sth like this:

```scala
Try{
    MyProtoClass.parseFrom(arrayOfBytes)
}.getOrElse{
    doSthElse
}
```

I intentionally use parseFrom instead of validate and this is why:
It seems that this line: 

```scala
MyProtoClass.parseFrom(arrayOfBytes)
```

somehow doesnt 'materialize' the object and exception is not thrown in Try block. I looked at the implementation and its implemented with some stream so even assigning this line to val and invoking val doesnt help. So exception is not thrown in Try{} block and somewhere further in the code I've got IndexOfOfBound exception, because somewhere later in the code I 'materialize' object from array[Byte] which I expected to fail but it didnt fail in Try{} where I wanted to fail.
So my stupid solution is somehow invoke every field of this parsed object in other way. One way is for example:

```scala
 Try{
    val hack = MyProtoClass.parseFrom(arrayOfBytes)
    hack.getAllFields
    hack
}.getOrElse{
    doSthElse
}
```
other way is println(hack),  so intuition tells me whatever works which will invoke all the fields of parsed object. It looks like a bug.

Btw it seems like its related with this issue:

https://github.com/scalapb/ScalaPB/issues/225

Version:
"com.trueaccord.scalapb" %% "scalapb-runtime" % "0.5.42" % "protobuf"

**Comment by thesamet** [comment]
Can you give an example of a proto and arrayOfBytes that cause this? What's the stack trace that you are getting? Is this issue different from #225?

**Comment by thesamet** [comment]
Closed via 47b4b33
